apps/web-giddh/src/app/daybook/daybook.component.ts

```diff
@@ -254,7 +267,77 @@ export class DaybookComponent implements OnInit, OnDestroy {
         this.destroyed$.complete();
     }
 
-    public expandEntry(entry) {
+    /**
+     * To check is entry expanded
+     *
+     * @param {*} entry Transaction object
+     * @memberof DaybookComponent
+     */
+    public expandEntry(entry): any {
+        let isInventory: boolean = false;
         entry.isExpanded = !entry.isExpanded;
+        if (entry && entry.otherTransactions) {
+            isInventory = entry.otherTransactions.some(otherTrasaction => {
+                if (otherTrasaction && otherTrasaction.inventory) {
+                    return true;
+                } else {
+                    return false;
+                }
+            });
+        }
+        if (isInventory && entry.isExpanded) {
+            this.checkIsStockEntryAvailable();
+        }
+    }
+
+    /**
+     *To check is there any stock entry available
+     *
+     * @memberof DaybookComponent
+     */
+    public checkIsStockEntryAvailable(): any {
+        this.daybookData$.subscribe(item => {
```

Comment:
`checkIsStockEntryAvailable` is called twice and there is no take or takeUntil every time it is called it will create a subscription

Reply:
This is not state data we are convert it in observable intentionally so no need to add